stdlib/public/core/StringUTF8.swift

```diff
@@ -174,10 +174,10 @@ extension String {
 
     init(_ _core: _StringCore) {
       self._core = _core
-      self._endIndex = Index(_core, _core.endIndex, Index._emptyBuffer)
+      self._endIndex = Index(_core.endIndex, Index._emptyBuffer)
```

Comment:
I actually think we don't need to store `_endIndex` anymore, it buys us nothing.  We can just compute it using this expression when we are asked for it.